core/net/mac/tsch/sample-project-conf.h

```diff
@@ -0,0 +1,199 @@
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ *         Sample project-file.h for Contiki TSCH
+ * \author
+ *         Simon Duquennoy <simonduq@sics.se>
+ *         Yasuyuki Tanaka <yasuyuki.tanaka@inf.ethz.ch>
+ *
+ * \brief
+ *         This file provide a sample project configuration using TSCH. Please
+ *         refer to README.md under core/net/mac/tsch for further information.
+ */
+
+#ifndef __PROJECT_CONF_H__
+#define __PROJECT_CONF_H__
+
+/*-----------------------------*/
+/* Project Local Configuration */
+/*-----------------------------*/
+/* Set 1 to WITH_SECURITY to enable TSCH security */
+#ifndef WITH_SECURITY
+#define WITH_SECURITY       0
+#endif /* WITH_SECURITY */
+
+/* TSCH logging. 0: disabled. 1: basic log. 2: with logs in interrupt. */
+#undef TSCH_LOG_CONF_LEVEL
+#define TSCH_LOG_CONF_LEVEL 2
+
+/* Do not start TSCH at init; needs to be started by an upper layer */
+#undef TSCH_CONF_AUTOSTART
+#define TSCH_CONF_AUTOSTART 0
+
+
+/*------------------------------------------*/
+/* IEEE 802.15.4/TSCH Network Configuration */
+/*------------------------------------------*/
+/* IEEE802.15.4 PANID */
+#undef IEEE802154_CONF_PANID
+#define IEEE802154_CONF_PANID 0xabcd
+
+/* 6TiSCH minimal schedule length.  The larger values, the less active slots */
+#undef TSCH_SCHEDULE_CONF_DEFAULT_LENGTH
+#define TSCH_SCHEDULE_CONF_DEFAULT_LENGTH 3
+
+
+/*---------------------------------*/
+/* Configuration Required for TSCH */
+/*---------------------------------*/
+#undef NETSTACK_CONF_MAC
+#define NETSTACK_CONF_MAC        tschmac_driver
+
+#undef NETSTACK_CONF_RDC
+#define NETSTACK_CONF_RDC        nordc_driver
+
+#undef NETSTACK_CONF_FRAMER
+#define NETSTACK_CONF_FRAMER     framer_802154
+
+#undef FRAME802154_CONF_VERSION
+#define FRAME802154_CONF_VERSION FRAME802154_IEEE802154E_2012
+
+
+/*------------------------*/
+/* Security Configuration */
+/*------------------------*/
+#if WITH_SECURITY
+#undef LLSEC802154_CONF_ENABLED
+#define LLSEC802154_CONF_ENABLED            1
+
+#undef LLSEC802154_CONF_USES_EXPLICIT_KEYS
+#define LLSEC802154_CONF_USES_EXPLICIT_KEYS 1
+
+#undef LLSEC802154_CONF_USES_FRAME_COUNTER
+#define LLSEC802154_CONF_USES_FRAME_COUNTER 0
+#endif /* WITH_SECURITY */
+
+
+/*-------------------*/
+/* RPL Configuration */
+/*-------------------*/
+#define RPL_CALLBACK_PARENT_SWITCH    tsch_rpl_callback_parent_switch
+#define RPL_CALLBACK_NEW_DIO_INTERVAL tsch_rpl_callback_new_dio_interval
+#define TSCH_CALLBACK_JOINING_NETWORK tsch_rpl_callback_joining_network
+#define TSCH_CALLBACK_LEAVING_NETWORK tsch_rpl_callback_leaving_network
+
+
+/*------------------------------------*/
+/* Z1 Platform Specific Configuration */
+/*------------------------------------*/
+#if CONTIKI_TARGET_Z1
```

Comment:
Honestly, I'm not sure about the sky platform nor z1... 😅  But, I can do that with adding `CONTIKI_TARGET_SKY`.